src/databases/PlainText/avtPlainTextFileFormat.C

```diff
@@ -1,633 +1,651 @@
-
-        if (nrows < 5)
-        {
-            // Check the first several rows to make sure it's ASCII.
-            if (!StringHelpers::IsPureASCII(buff, linelen))
-                EXCEPTION2(InvalidFilesException, filename.c_str(), "Not ASCII.");
-        }
-        else
-        {
-#ifdef MDSERVER
-            // We don't need to know how many rows there are in the
-            // MDServer, nor do we need to read all the data.  We do
-            // want to make sure there's at least one row of data,
-            // though, because if not, its probably not even a plain
-            // text file.  And we probably don't want to break out
-            // until we've got a good clue this file is truly ASCII.
-            break;
-#endif
-        }
-    }
-    delete [] buff;
-
-    if (ncolumns == 0 || (!firstRowIsHeader && nrows == 0))
-    {
-        EXCEPTION2(InvalidFilesException, filename.c_str(),
-                   "Couldn't parse anything meaningful from the file.");
-    }
-
-    // fix up variable names
-    if (format == Grid)
-    {
-        // we need exactly one variable name for 2D grid files
-        if (variableNames.size() > 1)
-            variableNames.resize(1);
-        if (variableNames.size() == 0)
-            variableNames.push_back("var");
-    }
-    else if (format == Columns)
-    {
-        // If we didn't get enough variable names, fill it out with
-        // generic values; and if it's too much, trim it
-        for (int i=(int)variableNames.size(); i<ncolumns; i++)
-        {
-            char str[20];
-            sprintf(str, "var%02d", i);
-            variableNames.push_back(str);
-        }
-        if (variableNames.size() > (size_t)ncolumns)
-            variableNames.resize(ncolumns);
-    }
-
-    //in().close();
-    fileRead = true;
-}
+// Copyright (c) Lawrence Livermore National Security, LLC and other VisIt
+// Project developers.  See the top-level LICENSE file for dates and other
+// details.  No copyright assignment is required to contribute to VisIt.
+
+// ************************************************************************* //
+//                            avtPlainTextFileFormat.C                           //
+// ************************************************************************* //
+
+#include <avtPlainTextFileFormat.h>
+
+#include <cstring>
+#include <string>
+
+#include <vtkPointData.h>
+#include <vtkDoubleArray.h>
+#include <vtkRectilinearGrid.h>
+#include <vtkUnstructuredGrid.h>
+#include <vtkPolyData.h>
+#include <vtkCellArray.h>
+#include <vtkVisItUtility.h>
+
+#include <avtDatabaseMetaData.h>
+
+#include <DBOptionsAttributes.h>
+#include <Expression.h>
+#include <StringHelpers.h>
+
+#include "visit_gzstream.h"
+
+#include <InvalidVariableException.h>
+#include <InvalidFilesException.h>
+
+
+using     std::string;
+using     std::vector;
+
+string
+TrimLeadingandTrailingSpaces(string str)
+{
+    auto start = str.find_first_not_of(" ");

```

Comment:
Do we want to be filtering out more than just spaces? Tabs as well? Other whitespace?

Reply:
I think we should filter out all whitespace.

Reply:
Hmmm, not line breaks, right? So just horizontal tabs and spaces?

<html><body>
<!--StartFragment-->

' ' | (0x20) | space (SPC)
-- | -- | --
'\t' | (0x09) | horizontal tab (TAB)
'\n' | (0x0a) | newline (LF)
'\v' | (0x0b) | vertical tab (VT)
'\f' | (0x0c) | feed (FF)
'\r' | (0x0d) | carriage return (CR)

<!--EndFragment-->
</body>
</html>